### DISCUSSION TITLE
How to model a circuit as terminating on provider's demarc switch port?

### DISCUSSION THREAD

**Discussion by JonathonReinhart** [netbox-community/netbox#5432] [Q&A]
*I actually just posted this to [Google Groups](https://groups.google.com/g/netbox-discuss/c/E7nxQMvlgmk) but I subsequently saw the note about GitHub Discussions. I really like this concept so I figured I'd cross-post here, too. (Plus the interface is clean, and I prefer Markdown.)*

---

I'm trying to model something simple but can't figure out the NetBox way to do it:

Our provider has a 5 GbE + 2 SFP port switch (MRV OS606) in our rack, which serves as the demarcation point. We have three circuits (one dedicated internet access, two metro Ethernet / E-Lines to other sites), each of which is demarcated / terminates on one of those switch ports. (Their fiber connects to one of the SFP ports, but I don't really need to model that.)

I've created the switch in NetBox and added the ports as Interfaces, like a normal switch. And I created the Circuits in NetBox.

How do I make it such that those Circuits terminate at those Interfaces?
- I *don't* believe I should connect the Circuit to the demarc switch Interface
  - Instead, it's my router that should be connected to that Interface via a Cable.
- I could forget about modeling the carrier's demarc switch, and instead connect my router to the Circuit, but that's not ideal:
  - I want to populate the slot in my rack and keep track of which port corresponds to which circuit
- I see that the Circuit termination has free-form patch panel and cross-connect IDs, but those don't form object associations
- I could instead model the demarc switch ports as Front Ports, and create dummy Rear Ports which connect to the Circuit (via Cables that don't actually exit)
  - This seems like the closest to reality, but seems a little strange

Naturally, I'd like to be able to cable trace:
```
edge router A ⬄ demarc switch A ⬄ E-Line circuit ⬄ demarc switch B ⬄ edge router B
```

How do you more experienced users model something like this?

Also noteworthy is that issue #4900 is to be resolved in the upcoming 2.10 release; I'm not sure if this will bring any relevant changes.

Thanks!

**Comment by jeremystretch** [comment]
> I could forget about modeling the carrier's demarc switch, and instead connect my router to the Circuit, but that's not ideal

IMO this is technically the correct solution, since the demarc equipment is not your responsibility: it comprises part of the circuit. (If it was physically located outside your rack/cage, you probably wouldn't have given it any thought.)

**Comment by candlerb** [comment]
> If it was physically located outside your rack/cage, you probably wouldn't have given it any thought.

It depends.  Sometimes the data centre gives you a dangling cross-connect, and the demarc is the plug on the end of that cable (and you never see what's on the other end).

In other cases though, they present a circuit to a patch-panel or an Ethernet Access Device which you yourself host in your own rack space, and therefore need to model - if only to show which rack units it occupies.  Sometimes the provider drops a switch into your rack, and multiple "circuits" are activated on different ports on that switch.  In these latter cases, the demarc is a *port* (socket), and you run your own cable into that port.

It seems to me that what the OP would like is to be able to associate a circuit termination to a device, in the same way as a frontport or rearport is associated with a device.  A circuit termination would then become something very similar to an interface.  It could also have a type (e.g. 1000baseT).  However the type and device could be optional, for the case where you're not modelling the provider's equipment.

For now, in order to have circuits terminations as physical ports on a device, the best option I can see is to create frontport-rearport pairs on the termination device, and run "cables" between the rearports and the circuit terminations.

**Reply by jeremystretch** [reply]
Regardless of where the demarc device is located, the circuit terminates to the customer's router interface. Anything beyond that is abstracted by treating the path as a circuit.

**Comment by candlerb** [comment]
A common situation is that the telco's EAD sits in my rack, and there is a cable (that I own, and that I can point to) which runs between my router and the EAD.

In Netbox I can't plug a circuit termination directly into a router interface.  Rather, I can create a cable with my router port at one end and the circuit termination at the other end.

This is valid as far as it goes - for example the cable can have the right length and colour - but the circuit termination is "dangling" in the sense that it has no location in space, and in particular it has no association with the EAD device (which I had to model, because it occupies 1U in my rack)

**Reply by jeremystretch** [reply]
Sorry, but can we try to keep replies inside their threads? At first glance it looks like you're providing several answers to the OP.
